From d7b73e8a9f1616d4b76d480bba0c34bcd193ccfa Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Mon, 27 Jul 2020 08:16:39 -0700
Subject: [PATCH 1467/1851] drm: xlnx: mixer: fix comparison of unsigned
 expression < 0

commit 5e417ef81e51e86830077977adc2c4e619649924 from
https://github.com/Xilinx/linux-xlnx.git

drivers/gpu/drm/xlnx/xlnx_mixer.c: In function 'xlnx_mix_disp_layer_enable':
drivers/gpu/drm/xlnx/xlnx_mixer.c:791:9: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
  791 |  if (id < XVMIX_LAYER_MASTER  || id > mixer_hw->logo_layer_id) {
...

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index 8621e53ac57b..8ba743e17fa0 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -788,7 +788,7 @@ static void xlnx_mix_disp_layer_enable(struct xlnx_mix_plane *plane)
 	mixer_hw = to_mixer_hw(plane);
 	l_data = plane->mixer_layer;
 	id = l_data->id;
-	if (id < XVMIX_LAYER_MASTER  || id > mixer_hw->logo_layer_id) {
+	if (id > mixer_hw->logo_layer_id) {
 		DRM_DEBUG_KMS("Attempt to activate invalid layer: %d\n", id);
 		return;
 	}
@@ -847,8 +847,7 @@ static void xlnx_mix_disp_layer_disable(struct xlnx_mix_plane *plane)
 	else
 		return;
 	layer_id = plane->mixer_layer->id;
-	if (layer_id < XVMIX_LAYER_MASTER  ||
-	    layer_id > mixer_hw->logo_layer_id)
+	if (layer_id > mixer_hw->logo_layer_id)
 		return;
 
 	xlnx_mix_layer_disable(mixer_hw, layer_id);
@@ -1292,7 +1291,7 @@ static int xlnx_mix_set_layer_scale(struct xlnx_mix_plane *plane,
 
 	if (!layer || !layer->hw_config.can_scale)
 		return -ENODEV;
-	if (val > XVMIX_SCALE_FACTOR_4X || val < XVMIX_SCALE_FACTOR_1X) {
+	if (val > XVMIX_SCALE_FACTOR_4X) {
 		DRM_ERROR("Mixer layer scale value illegal.\n");
 		return -EINVAL;
 	}
@@ -1369,7 +1368,7 @@ static int xlnx_mix_disp_set_layer_alpha(struct xlnx_mix_plane *plane,
 
 	if (!layer || !layer->hw_config.can_alpha)
 		return -ENODEV;
-	if (val > XVMIX_ALPHA_MAX || val < XVMIX_ALPHA_MIN) {
+	if (val > XVMIX_ALPHA_MAX) {
 		DRM_ERROR("Mixer layer alpha dts value illegal.\n");
 		return -EINVAL;
 	}
-- 
2.31.1

